src/screenshot.rs

```diff
@@ -166,24 +159,46 @@ impl Screenshot {
         &self,
         outputs: Vec<(wl_output::WlOutput, (i32, i32), String)>,
         app_id: &str,
-    ) -> anyhow::Result<HashMap<String, Arc<RgbaImage>>> {
+    ) -> anyhow::Result<HashMap<String, (u32, u32, Bytes)>> {
         // collect screenshots from each output
 
         let wayland_helper = self.wayland_helper.clone();
 
         let mut map = HashMap::with_capacity(outputs.len());
         for (output, _, name) in outputs {
             let frame = wayland_helper
-                .capture_source_shm(CaptureSource::Output(output), false)
+                .capture_source_shm(CaptureSource::Output(output.clone()), false)
                 .await
                 .ok_or_else(|| anyhow::anyhow!("shm screencopy failed"))?;
-            map.insert(name, Arc::new(frame.image()?));
+            map.insert(
+                name,
+                frame.image().map(|mut img| {
+                    img = match wayland_helper.output_info(&output) {
+                        Some(OutputInfo {
+                            transform: Transform::_90,
+                            ..
+                        }) => imageops::rotate90(&img),
+                        Some(OutputInfo {
+                            transform: Transform::_180,
+                            ..
+                        }) => imageops::rotate180(&img),
+                        Some(OutputInfo {
+                            transform: Transform::_270,
+                            ..
+                        }) => imageops::rotate270(&img),
+                        _ => img,
+                    };
+                    dbg!(img.width(), img.height());
```

Comment:
Presumably the `dbg!` call wasn't mean to be committed, and should be removed. (Or changed to something like a `log::debug!` with some more descriptive text).